vaadin-crud-flow/src/main/java/com/vaadin/flow/component/crud/Crud.java

```diff
@@ -187,8 +192,12 @@ private void registerHandlers() {
                 ((ComponentEventListener<CancelEvent<E>>) e -> {
                     cancelListeners.forEach(listener -> listener.onComponentEvent(e));
 
-                    setOpened(false);
-                    getEditor().clear();
+                    if ((this.gridActiveItem != null && this.getEditor().getItem() == this.gridActiveItem) || this.gridActiveItem == null) {
```

Comment:
Nit: would be good if this is broken down to several lines, so it will be easier to read.

Reply:
Removed.

Reply:
There is some minor problem with the formatting, but I guess it's good to go.